numba/tests/test_setimpl.py

```diff
@@ -0,0 +1,271 @@
+"""
+Testing C implementation of the numba sets
+"""
+
+import ctypes
+
+from numba.tests.support import TestCase
+from numba import _helperlib
+
+
+ENTRY_PRESENT = 1
+OK = 0
+ERR_KEY_NOT_FOUND = -1
+ERR_SET_MUTATED = -2
+ERR_ITER_EXHAUSTED = -3
+ERR_SET_EMPTY = -4
+ERR_CMP_FAILED = -5
+
+
+def to_bytes(key):
+    if isinstance(key, str):
+        key_bytes = bytes(key.encode())
+    else:
+        key_bytes = bytes(key)
```

Comment:
If `key` is an `int`, this will just allocate a zero-filled byte string of 3 elements